server/accounts.go

```diff
@@ -126,16 +146,22 @@ func (a *Account) AddServiceExport(subject string, accounts []*Account) error {
 		return ErrMissingAccount
 	}
 	if a.exports.services == nil {
-		a.exports.services = make(map[string]map[string]*Account)
+		a.exports.services = make(map[string]*exportAuth)
 	}
-	ma := a.exports.services[subject]
-	if accounts != nil && ma == nil {
-		ma = make(map[string]*Account)
-	}
-	for _, a := range accounts {
-		ma[a.Name] = a
+	ea := a.exports.services[subject]
+	if accounts != nil && ea == nil {
```

Comment:
Setting to nil ok since that indicates something, when we retrieve we check ok not for v != nil.

Reply:
But what I meant is that we populate `ea` only once, when it was not found and we create/populate it. But next time there is a lookup for subject and it is not nil, then nothing happens. Is that right?

Reply:
When we fetch ea it could be missing and nil, but we set it outside the loop regardless because if it was not found but we have no accounts[] then we need to set the entry to nil in the map. On authorization we check `v, ok := a.exports.services[subject]; !ok`

Reply:
Again, what I was asking is that: will there ever be an incremental call to AddServiceExport for the same account and same subject but possibly different list of accounts? If not, then that's fine, otherwise, I think there may be a problem.

Reply:
Meaning you want the function to treat the authorized list as idempotent correct?

Reply:
I want to know if there will be a case of calling AddServicesExport() with same account/subject and different list of accounts. If not, then code is fine. Otherwise, not sure.
I see that in one case this function is called from a place where we just create the account (buildInternal) so that's fine, but otherwise it looks like it is from an update of an account.